### PR TITLE
Patterns Singleton

### PR DESCRIPTION
**Pull request by AntonTmsh**
4.5. Singleton

### CODE REVIEW

File: Singleton/ThreadSafeSingletonWithLock.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Epam.NetMentoring.Singleton
+{
+    class ThreadSafeSingletonWithLock
```
Comment (fojman) [inline Singleton/ThreadSafeSingletonWithLock.cs]: don't forget about access modifiers, for clarity better to use them everywhere

**Comment by fojman** [comment]
Minor comments:
- don't miss access modifiers
- use auto formatting for all code
- write proper test cases

Overall looks good!

**Review by fojman — APPROVED** [review]
Approving, but please take a look at my comment

File: Singleton/ThreadSafeSingletonWithLock.cs (hunk 1 above)
Comment (AntonTmsh) [reply Singleton/ThreadSafeSingletonWithLock.cs]: ok

**Pull request merged**